aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -337,6 +337,50 @@ Tensor& cauchy_impl_(Tensor& self, double median, double sigma, c10::optional<Ge
   return self;
 }
 
+// ==================================================== Bernoulli =====================================================
+
+template<template<typename> class bernoulli_tensor_kernel, typename RNG>
+Tensor& bernoulli_impl_(Tensor& self, const Tensor& p_, c10::optional<Generator> gen) {
+  NoNamesGuard guard;
+  bernoulli_tensor_kernel<RNG>()(self, p_, gen);
+  return self;
+}
+
+template<template<typename> class bernoulli_scalar_kernel, typename RNG>
+Tensor& bernoulli_impl_(Tensor& self, double p, c10::optional<Generator> gen) {
+  TORCH_CHECK(0 <= p && p <= 1, "bernoulli_ expects p to be in [0, 1], but got p=", p);
+  bernoulli_scalar_kernel<RNG>()(self, p, gen);
+  return self;
+}
+
+template<template<typename> class bernoulli_tensor_kernel, typename RNG>
+Tensor bernoulli_impl(const Tensor& self, c10::optional<Generator> gen) {
+  Tensor result = at::empty_like(self, MemoryFormat::Contiguous);
+  bernoulli_impl_<bernoulli_tensor_kernel, RNG>(result, self, gen);
+  return result;
+}
+
+template<template<typename> class bernoulli_scalar_kernel, typename RNG>
+Tensor bernoulli_impl(const Tensor& self, double p, c10::optional<Generator> gen) {
+  Tensor result = at::empty_like(self, MemoryFormat::Contiguous);
+  //  bernoulli_impl_<bernoulli_scalar_kernel, RNG>(result, p, gen);
```

Comment:
It is quite weird that handling `bernoulli_impl_<bernoulli_scalar_kernel, RNG>(result, p, gen)` here would cause a `Segmentation fault` in RNG test during the CI process.
It can be solved by handling `result.bernoulli_(p, gen)` as a instead , but I have no ideal of what different between this two handle way.
The comment in `native_functions.yaml` says `bernoulli.p()` isn't used explicitly, I'm not sure does it related.
Please correct me, if I have missed something. orz

Reply:
Honestly I don't know the answer to this problem... I'm going to create a copy of your branch to run on CI and try to debug it there

https://github.com/pytorch/pytorch/pull/39429